pkg/apis/core/validation/shoot.go

```diff
@@ -2887,90 +2915,125 @@ func ValidateCoreDNSRewritingCommonSuffixes(commonSuffixes []string, fldPath *fi
 	)
 
-	if operation == "" && maintenanceOperation == "" {
+	if len(operations) == 0 && len(maintenanceOperations) == 0 {
+		return allErrs
+	}
+
+	if len(strings.Join(operations, ";")) > maxOperationsSize {
+		allErrs = append(allErrs, field.TooLong(fldPathOp, len(operations), maxOperationsSize))
+	}
+	if len(strings.Join(maintenanceOperations, ";")) > maxOperationsSize {
+		allErrs = append(allErrs, field.TooLong(fldPathMaintOp, len(maintenanceOperations), maxOperationsSize))
+	}
+	if len(allErrs) > 0 {
 		return allErrs
 	}
 
 	for _, res := range shoot.Status.EncryptedResources {
 		encryptedResources.Insert(schema.ParseGroupResource(res))
 	}
 
-	fldPathOp := fldPath.Key(v1beta1constants.GardenerOperation)
-	fldPathMaintOp := fldPath.Key(v1beta1constants.GardenerMaintenanceOperation)
+	var (
+		operationsSet            = sets.New(operations...)
+		maintenanceOperationsSet = sets.New(maintenanceOperations...)
+	)
 
-	if operation == maintenanceOperation {
-		allErrs = append(allErrs, field.Forbidden(fldPath, fmt.Sprintf("annotations %s and %s must not be equal", fldPathOp, fldPathMaintOp)))
+	if operationsSet.HasAny(maintenanceOperations...) {
+		allErrs = append(allErrs, field.Forbidden(fldPath, fmt.Sprintf("annotations %s and %s must not have any equal operations", fldPathOp, fldPathMaintOp)))
 	}
 
-	if operation != "" {
-		if forbiddenETCDEncryptionKeyShootOperationsWithK8s134.Has(operation) && !k8sLess134 {
-			allErrs = append(allErrs, field.Forbidden(fldPathOp, fmt.Sprintf("for Kubernetes versions >= 1.34, operation '%s' is no longer supported, please use 'rotate-etcd-encryption-key' instead, which performs a complete etcd encryption key rotation", operation)))
+	for _, op := range operationsSet.UnsortedList() {
+		if forbiddenETCDEncryptionKeyShootOperationsWithK8s134.Has(op) && !k8sLess134 {
+			allErrs = append(allErrs, field.Forbidden(fldPathOp, fmt.Sprintf("for Kubernetes versions >= 1.34, operation '%s' is no longer supported, please use 'rotate-etcd-encryption-key' instead, which performs a complete etcd encryption key rotation", op)))
 		}
-		if !availableShootOperations.Has(operation) && !strings.HasPrefix(operation, v1beta1constants.OperationRotateRolloutWorkers) {
-			allErrs = append(allErrs, field.NotSupported(fldPathOp, operation, sets.List(availableShootOperations)))
+		if !availableShootOperations.Has(op) && !strings.HasPrefix(op, v1beta1constants.OperationRotateRolloutWorkers) {
+			allErrs = append(allErrs, field.NotSupported(fldPathOp, op, sets.List(availableShootOperations)))
+		} else if len(operations) > 1 && !availableShootOperationsToRunInParallel.Has(op) && !strings.HasPrefix(op, v1beta1constants.OperationRotateRolloutWorkers) {
+			allErrs = append(allErrs, field.Forbidden(fldPathOp, fmt.Sprintf("operation '%s' is not permitted to be run in parallel with other operations", op)))
+		}
+		if forbiddenOps, ok := forbiddenShootOperationsToRunTogether[op]; ok && forbiddenOps.HasAny(operations...) {
```

Comment:
Minor optimization:
```suggestion
		if forbiddenOps, ok := forbiddenShootOperationsToRunTogether[op]; ok && operationsSet.HasAny(forbiddenOps...) {
```

forbiddenOps could be changed to a slice I guess which also helps with error messages below.